### PR TITLE
Add distance calculation from clinical-trial-matching-engine

### PR DESCRIPTION
**Pull request by jtquach1**
Changes made
- Adapted logic to get distance from search query zip code to closest study site from https://github.com/mcode/clinical-trial-matching-engine/blob/6fa5e8ec948052282eafe7876675debaf44ec1d5/src/app/services/ResearchStudySearchEntry.ts#L268-L310 and https://github.com/mcode/clinical-trial-matching-engine/blob/6fa5e8ec948052282eafe7876675debaf44ec1d5/src/app/services/distance.service.ts#L15-L23
- Adapted test bundle entries from https://github.com/mcode/clinical-trial-matching-engine/blob/6fa5e8ec948052282eafe7876675debaf44ec1d5/src/app/services/ResearchStudySearchEntry.spec.ts#L19-L194 and tests from https://github.com/mcode/clinical-trial-matching-engine/blob/6fa5e8ec948052282eafe7876675debaf44ec1d5/src/app/services/distance.service.spec.ts#L17-L37
- Added geolib and us-zips as dependencies

### CODE REVIEW

**Comment by jtquach1** [comment]
# Screenshots of visual changes (compared to #8)
| Before | After |
| - | - |
| image | image |

**Comment by jtquach1** [comment]
Just squashed the two commits to make rebasing #12 easier

**Comment by ChristineDuong** [comment]
Great translation of the distance functionality from the engine! The only thing is that I think that the distance calculation should be moved to the "backend" where the query sits. This is because Next will cache the results for us and theoretically, since the list of ResearchStudies will not change, the distances under this zipcode (or general query) for the closest facility will also not change. 

So if someone were to run the same search again for the patient, even though the results are the same, at the moment it'll recalculate all of the distances again. If we move it to be incorporated to the query part it should be cached together.

File: src/components/Results/Results.tsx [hunk 1]
```diff
@@ -1,42 +1,37 @@
...
+      matching trials...
+    </Typography>
+
+    {entries.map((entry: BundleEntry, index: number) => (
+      <Study
+        key={entry.resource.id}
+        entry={entry}
+        handleSaveStudy={handleSaveStudy(entry)}
+        isStudySaved={state.has(entry.resource.id)}
+        closestFacility={closestFacilities[index]}
```
Comment (ChristineDuong) [inline src/components/Results/Results.tsx:31]: Writing a comment here for posterity more than asking for changes.

This does achieve what is needed for the ticket, but indexing through a secondary array will make it hard to keep the entries and their distances linked together for things such as sorting and filtering. That being said -- those features don't exist and I think that this makes sense given the way the Results/Study components expect a Bundle entry/ResearchStudy which are FHIR defined. I think these things will also be resolved in PDM-538 so I'm okay with this.

File: src/pages/api/clinical-trial-search.ts [hunk 2]
```diff
@@ -32,6 +54,9 @@ const handler = async (req: NextApiRequest, res: NextApiResponse): Promise<void>
       ? searchParams.matchingServices
       : [searchParams.matchingServices];
   const results = await callWrappers(chosenServices, patientBundle);
+
+  console.log('results', results);
```
Comment (ChristineDuong) [inline src/pages/api/clinical-trial-search.ts:58]: This should be removed.

File: src/components/Results/Results.tsx (hunk 1 above)
Comment (jtquach1) [reply src/components/Results/Results.tsx:31]: For the future: I could refactor the results into some Record<string, Object> where the string is the id of a ResearchStudy and the Object has this structure: `{entry: BundleEntry, contactProps: ContactProps}` just so the ContactProps remains together with its corresponding ResearchStudy (inside the BundleEntry)

**Review by ChristineDuong — APPROVED** [review]

**Pull request merged**